optaplanner-integration/optaplanner-spring-boot-autoconfigure/src/main/java/org/optaplanner/spring/boot/autoconfigure/OptaPlannerProperties.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.spring.boot.autoconfigure;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+
+@ConfigurationProperties(value = "optaplanner", ignoreUnknownFields = false)
+public class OptaPlannerProperties {
+
+    public static final String DEFAULT_SOLVER_CONFIG_URL = "solverConfig.xml";
+
+    /**
+     * A classpath resource to read the solver configuration XML.
+     * Defaults to {@value DEFAULT_SOLVER_CONFIG_URL}.
+     * If this property isn't specified, that solverConfig.xml is optional.
+     */
+    private String solverConfigXML;
+
+    private SolverProperties solver;
+    private SolverManagerProperties solverManager;
```

Comment:
Should be annotated with `@NestedConfigurationProperty`:
https://docs.spring.io/spring-boot/docs/current/reference/htmlsingle/#configuration-metadata-nested-properties